In [1]:
import pandas as pd
import numpy as np

import warnings
warnings.filterwarnings(action='ignore')
import tensorflow as tf

import math
import matplotlib.pyplot as plt
from keras import regularizers
from keras.models import Sequential
from keras.layers import Dense, LSTM, Dropout
from sklearn.preprocessing import MinMaxScaler
from sklearn.metrics import mean_squared_error
import pickle
import gzip
import tensorflow as tf
# tf.random.set_seed(1004)
tf.random.set_seed(2)

gpus = tf.config.experimental.list_physical_devices('GPU')
print(gpus)
if gpus:
    try:
        for i in range(len(gpus)):
            tf.config.experimental.set_memory_growth(gpus[i], True)
    except RuntimeError as e:
        # 프로그램 시작시에 메모리 증가가 설정되어야만 합니다
        print(e)

[PhysicalDevice(name='/physical_device:GPU:0', device_type='GPU'), PhysicalDevice(name='/physical_device:GPU:1', device_type='GPU')]


# Data Load

In [2]:
import random  
seed_num = 42
random.seed(seed_num)

x = np.load('/project/LSH/x_(7727,10,4068).npy')
y = np.load('/project/LSH/y_(7727,1).npy')

idx = list(range(len(x)))
random.shuffle(idx)

i = round(x.shape[0]*0.8)
X_train, y_train = x[idx[:i],:,:], y[idx[:i]]
X_test, y_test = x[idx[i:],:,:], y[idx[i:]]

X_train.shape, y_train.shape, X_test.shape, y_test.shape

((6182, 10, 4068), (6182,), (1545, 10, 4068), (1545,))

In [3]:
def MAE(y_true, y_pred):
    return "{:.2e}".format(np.mean(np.abs(y_true-y_pred)))

def MSE(y_true, y_pred):
    return "{:.2e}".format(np.mean(np.square(y_true-y_pred)))

def RMSE(y_true, y_pred):
    return "{:.2e}".format(np.sqrt(np.mean(np.square(y_true-y_pred))))

In [4]:
from sklearn.ensemble import AdaBoostRegressor, AdaBoostClassifier
from keras.wrappers.scikit_learn import KerasClassifier, KerasRegressor
from sklearn.ensemble import VotingClassifier
from keras.layers import InputLayer, GRU
import keras

# Modeling

## LSTM 정의

In [5]:
tf.random.set_seed(seed_num)
# def get_model():
#     lstm = Sequential()
#     lstm.add(InputLayer(input_shape=(x.shape[1],x.shape[2])))
#     lstm.add(LSTM(units=128, activation='hard_sigmoid', return_sequences=True))
#     lstm.add(LSTM(units=64, activation='hard_sigmoid', return_sequences=True))
#     lstm.add(Dropout(0.2))
#     lstm.add(LSTM(units=64, activation='hard_sigmoid', return_sequences=True))
#     lstm.add(LSTM(units=32, activation='hard_sigmoid', return_sequences=False))
#     lstm.add(Dropout(0.2))
#     lstm.add(Dense(units=1, activation='sigmoid'))
#     lstm.compile(optimizer= keras.optimizers.Adam(learning_rate = 0.001), 
#                           loss = "binary_crossentropy", metrics=['acc'])
#     return lstm
def get_model():
    lstm = Sequential()
    lstm.add(InputLayer(input_shape=(X_train.shape[1],X_train.shape[2])))
    lstm.add(LSTM(units=128, activation='relu', return_sequences=True, kernel_regularizer=regularizers.l2(0.001)))
    lstm.add(LSTM(units=64, activation='relu', return_sequences=True, kernel_regularizer=regularizers.l2(0.001)))
    lstm.add(LSTM(units=32, activation='tanh', return_sequences=False, kernel_regularizer=regularizers.l2(0.001)))
    lstm.add(Dense(units=1))

    lstm.compile(optimizer= keras.optimizers.Adam(learning_rate = 0.001), 
                          loss = "binary_crossentropy", metrics=['acc'])
    return lstm

In [6]:
# lstm.fit(X_train,y_train, epochs=1000, batch_size=256, validation_split = 0.2)

# #예측 후 가공
# preds = lstm.predict(X_test)

# preds[preds>0.5]=1
# preds[preds<=0.5]=0

# from sklearn import metrics 
# print('정확도 :', metrics.accuracy_score(y_test, preds))
# print("RMSE : ", RMSE(y_test, preds))

## VotingClassifier 적용

In [15]:
%%time
#LSTM 쌓기
with tf.device('/device:GPU:1'):
    estimator = []
    for i in range(1,51):
        LSTM_Predictors = KerasClassifier(build_fn=lambda:get_model(), epochs=100, batch_size=256)
        LSTM_Predictors._estimator_type="classifier"
        estimator.append((f'model{i}', LSTM_Predictors))
    print(estimator)    

    final_model = VotingClassifier(estimators = estimator, voting = 'soft')
    final_model.fit(X_train,y_train)

    #예측 후 가공
    preds = final_model.predict(X_test)

    preds[preds>0.5]=1
    preds[preds<=0.5]=0

    precision = precision_score(y_test, preds)
    recall = recall_score(y_test, preds)
    f1 = f1_score(y_test, preds)
    roc_auc = roc_auc_score(y_test, preds)
    acc = accuracy_score(y_test, preds)

    print(f' accuracy : {acc}, precision : {precision}, recall : {recall}, f1 : {f1}, roc_auc : {roc_auc}')

[('model1', <tensorflow.python.keras.wrappers.scikit_learn.KerasClassifier object at 0x7f3800590b50>), ('model2', <tensorflow.python.keras.wrappers.scikit_learn.KerasClassifier object at 0x7f38004b8280>), ('model3', <tensorflow.python.keras.wrappers.scikit_learn.KerasClassifier object at 0x7f38004b8250>), ('model4', <tensorflow.python.keras.wrappers.scikit_learn.KerasClassifier object at 0x7f38004b8220>), ('model5', <tensorflow.python.keras.wrappers.scikit_learn.KerasClassifier object at 0x7f38004b8790>), ('model6', <tensorflow.python.keras.wrappers.scikit_learn.KerasClassifier object at 0x7f38004b8310>), ('model7', <tensorflow.python.keras.wrappers.scikit_learn.KerasClassifier object at 0x7f38004b8370>), ('model8', <tensorflow.python.keras.wrappers.scikit_learn.KerasClassifier object at 0x7f38004b8340>), ('model9', <tensorflow.python.keras.wrappers.scikit_learn.KerasClassifier object at 0x7f38004b8c10>), ('model10', <tensorflow.python.keras.wrappers.scikit_learn.KerasClassifier object

25/25 [==============================] - 1s 29ms/step - loss: 0.3783 - acc: 0.9329
Epoch 30/100
25/25 [==============================] - 1s 29ms/step - loss: 0.5017 - acc: 0.8766
Epoch 31/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3891 - acc: 0.9340
Epoch 32/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3429 - acc: 0.9471
Epoch 33/100
25/25 [==============================] - 1s 30ms/step - loss: 0.3271 - acc: 0.9478
Epoch 34/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3146 - acc: 0.9534
Epoch 35/100
25/25 [==============================] - 1s 30ms/step - loss: 0.3179 - acc: 0.9537
Epoch 36/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3240 - acc: 0.9529
Epoch 37/100
25/25 [==============================] - 1s 29ms/step - loss: 0.6726 - acc: 0.8983
Epoch 38/100
25/25 [==============================] - 1s 29ms/step - loss: 0.4199 - acc: 0.9313
Epoch 39/100
25/25 [==============================] -

25/25 [==============================] - 1s 31ms/step - loss: 0.5482 - acc: 0.8687
Epoch 12/100
25/25 [==============================] - 1s 30ms/step - loss: 0.5336 - acc: 0.8895
Epoch 13/100
25/25 [==============================] - 1s 30ms/step - loss: 0.9764 - acc: 0.8418
Epoch 14/100
25/25 [==============================] - 1s 30ms/step - loss: 0.7035 - acc: 0.8195
Epoch 15/100
25/25 [==============================] - 1s 30ms/step - loss: 0.5757 - acc: 0.8675
Epoch 16/100
25/25 [==============================] - 1s 30ms/step - loss: 0.5392 - acc: 0.8897
Epoch 17/100
25/25 [==============================] - 1s 30ms/step - loss: 0.4953 - acc: 0.9080
Epoch 18/100
25/25 [==============================] - 1s 30ms/step - loss: 0.4484 - acc: 0.9249
Epoch 19/100
25/25 [==============================] - 1s 30ms/step - loss: 0.4261 - acc: 0.9329
Epoch 20/100
25/25 [==============================] - 1s 30ms/step - loss: 0.4165 - acc: 0.9355
Epoch 21/100
25/25 [==============================] -

25/25 [==============================] - 1s 27ms/step - loss: 0.3479 - acc: 0.9634
Epoch 97/100
25/25 [==============================] - 1s 28ms/step - loss: 0.3233 - acc: 0.9696
Epoch 98/100
25/25 [==============================] - 1s 28ms/step - loss: 0.2938 - acc: 0.9743
Epoch 99/100
25/25 [==============================] - 1s 27ms/step - loss: 0.2803 - acc: 0.9782
Epoch 100/100
25/25 [==============================] - 1s 27ms/step - loss: 0.2680 - acc: 0.9809
Epoch 1/100
25/25 [==============================] - 3s 29ms/step - loss: 1.8649 - acc: 0.5147
Epoch 2/100
25/25 [==============================] - 1s 30ms/step - loss: 0.9698 - acc: 0.6194
Epoch 3/100
25/25 [==============================] - 1s 30ms/step - loss: 0.8130 - acc: 0.6902
Epoch 4/100
25/25 [==============================] - 1s 30ms/step - loss: 0.7373 - acc: 0.7378
Epoch 5/100
25/25 [==============================] - 1s 29ms/step - loss: 0.6758 - acc: 0.7724
Epoch 6/100
25/25 [==============================] - 1s 2

25/25 [==============================] - 1s 29ms/step - loss: 0.2486 - acc: 0.9838
Epoch 79/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2585 - acc: 0.9775
Epoch 80/100
25/25 [==============================] - 1s 28ms/step - loss: 0.2450 - acc: 0.9786
Epoch 81/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2283 - acc: 0.9861
Epoch 82/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2195 - acc: 0.9867
Epoch 83/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2141 - acc: 0.9867
Epoch 84/100
25/25 [==============================] - 1s 28ms/step - loss: 0.2104 - acc: 0.9877
Epoch 85/100
25/25 [==============================] - 1s 28ms/step - loss: 0.3928 - acc: 0.9489
Epoch 86/100
25/25 [==============================] - 1s 28ms/step - loss: 0.2756 - acc: 0.9746
Epoch 87/100
25/25 [==============================] - 1s 28ms/step - loss: 0.3193 - acc: 0.9620
Epoch 88/100
25/25 [==============================] -

25/25 [==============================] - 1s 30ms/step - loss: 0.5413 - acc: 0.8845
Epoch 61/100
25/25 [==============================] - 1s 29ms/step - loss: 0.4558 - acc: 0.9251
Epoch 62/100
25/25 [==============================] - 1s 29ms/step - loss: 0.4314 - acc: 0.9461
Epoch 63/100
25/25 [==============================] - 1s 29ms/step - loss: 0.4304 - acc: 0.9518
Epoch 64/100
25/25 [==============================] - 1s 30ms/step - loss: 0.7552 - acc: 0.8449
Epoch 65/100
25/25 [==============================] - 1s 29ms/step - loss: 0.5762 - acc: 0.8745
Epoch 66/100
25/25 [==============================] - 1s 28ms/step - loss: 0.5063 - acc: 0.9138
Epoch 67/100
25/25 [==============================] - 1s 30ms/step - loss: 0.4438 - acc: 0.9351
Epoch 68/100
25/25 [==============================] - 1s 30ms/step - loss: 0.3997 - acc: 0.9529
Epoch 69/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3750 - acc: 0.9605
Epoch 70/100
25/25 [==============================] -

25/25 [==============================] - 1s 29ms/step - loss: 0.2615 - acc: 0.9749
Epoch 43/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2624 - acc: 0.9746
Epoch 44/100
25/25 [==============================] - 1s 28ms/step - loss: 0.3980 - acc: 0.9589
Epoch 45/100
25/25 [==============================] - 1s 28ms/step - loss: 0.4791 - acc: 0.8423
Epoch 46/100
25/25 [==============================] - 1s 29ms/step - loss: 0.5428 - acc: 0.8488
Epoch 47/100
25/25 [==============================] - 1s 29ms/step - loss: 0.4504 - acc: 0.9060
Epoch 48/100
25/25 [==============================] - 1s 28ms/step - loss: 0.4127 - acc: 0.9236
Epoch 49/100
25/25 [==============================] - 1s 28ms/step - loss: 0.9271 - acc: 0.8465
Epoch 50/100
25/25 [==============================] - 1s 28ms/step - loss: 0.6399 - acc: 0.8504
Epoch 51/100
25/25 [==============================] - 1s 28ms/step - loss: 0.4981 - acc: 0.9054
Epoch 52/100
25/25 [==============================] -

25/25 [==============================] - 1s 28ms/step - loss: 0.3350 - acc: 0.9478
Epoch 25/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3398 - acc: 0.9431
Epoch 26/100
25/25 [==============================] - 1s 30ms/step - loss: 0.3029 - acc: 0.9542
Epoch 27/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3678 - acc: 0.9453
Epoch 28/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3656 - acc: 0.9390
Epoch 29/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3307 - acc: 0.9529
Epoch 30/100
25/25 [==============================] - 1s 30ms/step - loss: 0.3003 - acc: 0.9583
Epoch 31/100
25/25 [==============================] - 1s 31ms/step - loss: 0.2878 - acc: 0.9634
Epoch 32/100
25/25 [==============================] - 1s 30ms/step - loss: 0.2949 - acc: 0.9646
Epoch 33/100
25/25 [==============================] - 1s 30ms/step - loss: 0.6063 - acc: 0.9028
Epoch 34/100
25/25 [==============================] -

25/25 [==============================] - 1s 29ms/step - loss: 0.5986 - acc: 0.8151
Epoch 7/100
25/25 [==============================] - 1s 29ms/step - loss: 0.6555 - acc: 0.7926
Epoch 8/100
25/25 [==============================] - 1s 29ms/step - loss: 0.5863 - acc: 0.8222
Epoch 9/100
25/25 [==============================] - 1s 29ms/step - loss: 0.5481 - acc: 0.8242
Epoch 10/100
25/25 [==============================] - 1s 28ms/step - loss: 0.6561 - acc: 0.7478
Epoch 11/100
25/25 [==============================] - 1s 29ms/step - loss: 0.5278 - acc: 0.8481
Epoch 12/100
25/25 [==============================] - 1s 29ms/step - loss: 0.4525 - acc: 0.8957
Epoch 13/100
25/25 [==============================] - 1s 29ms/step - loss: 0.5824 - acc: 0.8410
Epoch 14/100
25/25 [==============================] - 1s 28ms/step - loss: 0.4410 - acc: 0.8889
Epoch 15/100
25/25 [==============================] - 1s 28ms/step - loss: 0.3888 - acc: 0.9146
Epoch 16/100
25/25 [==============================] - 1s

25/25 [==============================] - 1s 29ms/step - loss: 5.9539 - acc: 0.6165
Epoch 92/100
25/25 [==============================] - 1s 29ms/step - loss: 5.8850 - acc: 0.6210
Epoch 93/100
25/25 [==============================] - 1s 30ms/step - loss: 5.8835 - acc: 0.6215
Epoch 94/100
25/25 [==============================] - 1s 29ms/step - loss: 5.8820 - acc: 0.6213
Epoch 95/100
25/25 [==============================] - 1s 29ms/step - loss: 5.8788 - acc: 0.6213
Epoch 96/100
25/25 [==============================] - 1s 28ms/step - loss: 5.8616 - acc: 0.6225
Epoch 97/100
25/25 [==============================] - 1s 28ms/step - loss: 5.8561 - acc: 0.6228
Epoch 98/100
25/25 [==============================] - 1s 28ms/step - loss: 5.8250 - acc: 0.6247
Epoch 99/100
25/25 [==============================] - 1s 28ms/step - loss: 5.7875 - acc: 0.6270
Epoch 100/100
25/25 [==============================] - 1s 28ms/step - loss: 5.7779 - acc: 0.6276
Epoch 1/100
25/25 [==============================] -

25/25 [==============================] - 1s 29ms/step - loss: 0.3261 - acc: 0.9612
Epoch 74/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2942 - acc: 0.9731
Epoch 75/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2684 - acc: 0.9744
Epoch 76/100
25/25 [==============================] - 1s 28ms/step - loss: 0.3078 - acc: 0.9736
Epoch 77/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2909 - acc: 0.9744
Epoch 78/100
25/25 [==============================] - 1s 28ms/step - loss: 0.2753 - acc: 0.9777
Epoch 79/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2933 - acc: 0.9738
Epoch 80/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2925 - acc: 0.9706
Epoch 81/100
25/25 [==============================] - 1s 28ms/step - loss: 0.2676 - acc: 0.9767
Epoch 82/100
25/25 [==============================] - 1s 28ms/step - loss: 0.2483 - acc: 0.9811
Epoch 83/100
25/25 [==============================] -

25/25 [==============================] - 1s 29ms/step - loss: 0.2695 - acc: 0.9728
Epoch 56/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2589 - acc: 0.9764
Epoch 57/100
25/25 [==============================] - 1s 28ms/step - loss: 0.2637 - acc: 0.9780
Epoch 58/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3700 - acc: 0.9554
Epoch 59/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3387 - acc: 0.9639
Epoch 60/100
25/25 [==============================] - 1s 28ms/step - loss: 0.2938 - acc: 0.9749
Epoch 61/100
25/25 [==============================] - 1s 28ms/step - loss: 0.2876 - acc: 0.9756
Epoch 62/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2736 - acc: 0.9801
Epoch 63/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2543 - acc: 0.9825
Epoch 64/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2547 - acc: 0.9843
Epoch 65/100
25/25 [==============================] -

25/25 [==============================] - 1s 28ms/step - loss: 0.3305 - acc: 0.9539
Epoch 38/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3248 - acc: 0.9555
Epoch 39/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3421 - acc: 0.9484
Epoch 40/100
25/25 [==============================] - 1s 28ms/step - loss: 0.4611 - acc: 0.9283
Epoch 41/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3968 - acc: 0.9426
Epoch 42/100
25/25 [==============================] - 1s 29ms/step - loss: 0.6785 - acc: 0.8763
Epoch 43/100
25/25 [==============================] - 1s 28ms/step - loss: 0.4565 - acc: 0.9314
Epoch 44/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3995 - acc: 0.9489
Epoch 45/100
25/25 [==============================] - 1s 34ms/step - loss: 0.3840 - acc: 0.9539
Epoch 46/100
25/25 [==============================] - 1s 29ms/step - loss: 0.4009 - acc: 0.9479
Epoch 47/100
25/25 [==============================] -

25/25 [==============================] - 1s 28ms/step - loss: 0.4738 - acc: 0.8991
Epoch 20/100
25/25 [==============================] - 1s 29ms/step - loss: 0.4266 - acc: 0.9232
Epoch 21/100
25/25 [==============================] - 1s 29ms/step - loss: 0.4236 - acc: 0.9319
Epoch 22/100
25/25 [==============================] - 1s 30ms/step - loss: 0.4128 - acc: 0.9280
Epoch 23/100
25/25 [==============================] - 1s 28ms/step - loss: 0.3939 - acc: 0.9380
Epoch 24/100
25/25 [==============================] - 1s 30ms/step - loss: 0.4517 - acc: 0.9324
Epoch 25/100
25/25 [==============================] - 1s 29ms/step - loss: 0.4197 - acc: 0.9351
Epoch 26/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3859 - acc: 0.9502
Epoch 27/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3646 - acc: 0.9573
Epoch 28/100
25/25 [==============================] - 1s 28ms/step - loss: 0.3419 - acc: 0.9621
Epoch 29/100
25/25 [==============================] -

25/25 [==============================] - 3s 34ms/step - loss: 1.8268 - acc: 0.5133
Epoch 2/100
25/25 [==============================] - 1s 29ms/step - loss: 0.9584 - acc: 0.6199
Epoch 3/100
25/25 [==============================] - 1s 28ms/step - loss: 0.7821 - acc: 0.7224
Epoch 4/100
25/25 [==============================] - 1s 28ms/step - loss: 0.7126 - acc: 0.7502
Epoch 5/100
25/25 [==============================] - 1s 29ms/step - loss: 0.6564 - acc: 0.7842
Epoch 6/100
25/25 [==============================] - 1s 28ms/step - loss: 0.7239 - acc: 0.7098
Epoch 7/100
25/25 [==============================] - 1s 29ms/step - loss: 0.6723 - acc: 0.7553
Epoch 8/100
25/25 [==============================] - 1s 29ms/step - loss: 0.6725 - acc: 0.7371
Epoch 9/100
25/25 [==============================] - 1s 30ms/step - loss: 0.6629 - acc: 0.7572
Epoch 10/100
25/25 [==============================] - 1s 28ms/step - loss: 0.5706 - acc: 0.8232
Epoch 11/100
25/25 [==============================] - 1s 29ms

25/25 [==============================] - 1s 29ms/step - loss: 0.3468 - acc: 0.9621
Epoch 87/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3703 - acc: 0.9623
Epoch 88/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3805 - acc: 0.9579
Epoch 89/100
25/25 [==============================] - 1s 30ms/step - loss: 0.4125 - acc: 0.9319
Epoch 90/100
25/25 [==============================] - 1s 29ms/step - loss: 0.4445 - acc: 0.9235
Epoch 91/100
25/25 [==============================] - 1s 30ms/step - loss: 0.3996 - acc: 0.9482
Epoch 92/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3514 - acc: 0.9621
Epoch 93/100
25/25 [==============================] - 1s 30ms/step - loss: 0.3266 - acc: 0.9673
Epoch 94/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3109 - acc: 0.9719
Epoch 95/100
25/25 [==============================] - 1s 30ms/step - loss: 0.3501 - acc: 0.9610
Epoch 96/100
25/25 [==============================] -

25/25 [==============================] - 1s 29ms/step - loss: 0.3643 - acc: 0.9511
Epoch 69/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2919 - acc: 0.9685
Epoch 70/100
25/25 [==============================] - 1s 30ms/step - loss: 0.2661 - acc: 0.9751
Epoch 71/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2488 - acc: 0.9782
Epoch 72/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2383 - acc: 0.9798
Epoch 73/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2304 - acc: 0.9798
Epoch 74/100
25/25 [==============================] - 1s 30ms/step - loss: 0.2238 - acc: 0.9803
Epoch 75/100
25/25 [==============================] - 1s 30ms/step - loss: 0.2179 - acc: 0.9820
Epoch 76/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2130 - acc: 0.9825
Epoch 77/100
25/25 [==============================] - 1s 30ms/step - loss: 0.2088 - acc: 0.9820
Epoch 78/100
25/25 [==============================] -

25/25 [==============================] - 1s 28ms/step - loss: 0.3012 - acc: 0.9712
Epoch 51/100
25/25 [==============================] - 1s 29ms/step - loss: 0.4749 - acc: 0.9431
Epoch 52/100
25/25 [==============================] - 1s 30ms/step - loss: 0.4586 - acc: 0.9447
Epoch 53/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3913 - acc: 0.9579
Epoch 54/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3941 - acc: 0.9523
Epoch 55/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3334 - acc: 0.9646
Epoch 56/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3086 - acc: 0.9738
Epoch 57/100
25/25 [==============================] - 1s 29ms/step - loss: 0.5172 - acc: 0.9385
Epoch 58/100
25/25 [==============================] - 1s 29ms/step - loss: 0.4329 - acc: 0.9316
Epoch 59/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3771 - acc: 0.9534
Epoch 60/100
25/25 [==============================] -

25/25 [==============================] - 1s 30ms/step - loss: 0.4451 - acc: 0.9368
Epoch 33/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3421 - acc: 0.9612
Epoch 34/100
25/25 [==============================] - 1s 28ms/step - loss: 0.3100 - acc: 0.9727
Epoch 35/100
25/25 [==============================] - 1s 28ms/step - loss: 0.2913 - acc: 0.9753
Epoch 36/100
25/25 [==============================] - 1s 28ms/step - loss: 0.2690 - acc: 0.9812
Epoch 37/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2839 - acc: 0.9778
Epoch 38/100
25/25 [==============================] - 1s 28ms/step - loss: 0.3346 - acc: 0.9665
Epoch 39/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2797 - acc: 0.9725
Epoch 40/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2547 - acc: 0.9801
Epoch 41/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2427 - acc: 0.9830
Epoch 42/100
25/25 [==============================] -

25/25 [==============================] - 1s 28ms/step - loss: 0.4081 - acc: 0.9254
Epoch 15/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3866 - acc: 0.9334
Epoch 16/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3641 - acc: 0.9410
Epoch 17/100
25/25 [==============================] - 1s 28ms/step - loss: 0.4730 - acc: 0.9261
Epoch 18/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3949 - acc: 0.9296
Epoch 19/100
25/25 [==============================] - 1s 28ms/step - loss: 0.3510 - acc: 0.9468
Epoch 20/100
25/25 [==============================] - 1s 28ms/step - loss: 0.4342 - acc: 0.9251
Epoch 21/100
25/25 [==============================] - 1s 29ms/step - loss: 0.4653 - acc: 0.9047
Epoch 22/100
25/25 [==============================] - 1s 30ms/step - loss: 0.4299 - acc: 0.9325
Epoch 23/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3898 - acc: 0.9350
Epoch 24/100
25/25 [==============================] -

25/25 [==============================] - 1s 29ms/step - loss: 0.1888 - acc: 0.9893
Epoch 100/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2139 - acc: 0.9887
Epoch 1/100
25/25 [==============================] - 3s 31ms/step - loss: 1.9799 - acc: 0.5196
Epoch 2/100
25/25 [==============================] - 1s 30ms/step - loss: 0.8966 - acc: 0.6313
Epoch 3/100
25/25 [==============================] - 1s 28ms/step - loss: 0.7508 - acc: 0.7413
Epoch 4/100
25/25 [==============================] - 1s 30ms/step - loss: 0.6637 - acc: 0.7821
Epoch 5/100
25/25 [==============================] - 1s 29ms/step - loss: 0.9243 - acc: 0.5631
Epoch 6/100
25/25 [==============================] - 1s 30ms/step - loss: 0.7356 - acc: 0.7143
Epoch 7/100
25/25 [==============================] - 1s 29ms/step - loss: 0.6617 - acc: 0.7818
Epoch 8/100
25/25 [==============================] - 1s 29ms/step - loss: 0.6216 - acc: 0.8052
Epoch 9/100
25/25 [==============================] - 1s 30ms

25/25 [==============================] - 1s 29ms/step - loss: 0.4331 - acc: 0.9463
Epoch 82/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3625 - acc: 0.9634
Epoch 83/100
25/25 [==============================] - 1s 28ms/step - loss: 0.3215 - acc: 0.9767
Epoch 84/100
25/25 [==============================] - 1s 29ms/step - loss: 0.9796 - acc: 0.8754
Epoch 85/100
25/25 [==============================] - 1s 30ms/step - loss: 0.6683 - acc: 0.8711
Epoch 86/100
25/25 [==============================] - 1s 30ms/step - loss: 0.5153 - acc: 0.9235
Epoch 87/100
25/25 [==============================] - 1s 29ms/step - loss: 0.4335 - acc: 0.9476
Epoch 88/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3870 - acc: 0.9625
Epoch 89/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3482 - acc: 0.9719
Epoch 90/100
25/25 [==============================] - 1s 30ms/step - loss: 0.3287 - acc: 0.9791
Epoch 91/100
25/25 [==============================] -

25/25 [==============================] - 1s 29ms/step - loss: 0.2348 - acc: 0.9838
Epoch 64/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2252 - acc: 0.9829
Epoch 65/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2165 - acc: 0.9858
Epoch 66/100
25/25 [==============================] - 1s 29ms/step - loss: 0.5917 - acc: 0.9338
Epoch 67/100
25/25 [==============================] - 1s 34ms/step - loss: 1.0824 - acc: 0.8601
Epoch 68/100
25/25 [==============================] - 1s 29ms/step - loss: 0.5632 - acc: 0.8926
Epoch 69/100
25/25 [==============================] - 1s 29ms/step - loss: 0.4391 - acc: 0.9330
Epoch 70/100
25/25 [==============================] - 1s 30ms/step - loss: 0.3760 - acc: 0.9505
Epoch 71/100
25/25 [==============================] - 1s 30ms/step - loss: 0.3520 - acc: 0.9573
Epoch 72/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3464 - acc: 0.9621
Epoch 73/100
25/25 [==============================] -

25/25 [==============================] - 1s 29ms/step - loss: 0.4159 - acc: 0.9410
Epoch 46/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3713 - acc: 0.9486
Epoch 47/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3213 - acc: 0.9597
Epoch 48/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2909 - acc: 0.9649
Epoch 49/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2934 - acc: 0.9643
Epoch 50/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2887 - acc: 0.9686
Epoch 51/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3173 - acc: 0.9641
Epoch 52/100
25/25 [==============================] - 1s 28ms/step - loss: 0.3307 - acc: 0.9597
Epoch 53/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2982 - acc: 0.9702
Epoch 54/100
25/25 [==============================] - 1s 30ms/step - loss: 0.2726 - acc: 0.9746
Epoch 55/100
25/25 [==============================] -

25/25 [==============================] - 1s 30ms/step - loss: 0.6675 - acc: 0.8303
Epoch 28/100
25/25 [==============================] - 1s 30ms/step - loss: 0.5894 - acc: 0.8374
Epoch 29/100
25/25 [==============================] - 1s 29ms/step - loss: 0.4607 - acc: 0.9114
Epoch 30/100
25/25 [==============================] - 1s 30ms/step - loss: 0.4123 - acc: 0.9259
Epoch 31/100
25/25 [==============================] - 1s 30ms/step - loss: 0.4502 - acc: 0.9130
Epoch 32/100
25/25 [==============================] - 1s 29ms/step - loss: 0.7431 - acc: 0.7049
Epoch 33/100
25/25 [==============================] - 1s 29ms/step - loss: 0.6386 - acc: 0.7946
Epoch 34/100
25/25 [==============================] - 1s 28ms/step - loss: 0.5137 - acc: 0.8609
Epoch 35/100
25/25 [==============================] - 1s 29ms/step - loss: 0.4983 - acc: 0.8717
Epoch 36/100
25/25 [==============================] - 1s 28ms/step - loss: 0.5390 - acc: 0.8227
Epoch 37/100
25/25 [==============================] -

25/25 [==============================] - 1s 28ms/step - loss: 0.8724 - acc: 0.5890
Epoch 10/100
25/25 [==============================] - 1s 29ms/step - loss: 0.6725 - acc: 0.7685
Epoch 11/100
25/25 [==============================] - 1s 29ms/step - loss: 0.9098 - acc: 0.5945
Epoch 12/100
25/25 [==============================] - 1s 29ms/step - loss: 0.7120 - acc: 0.7333
Epoch 13/100
25/25 [==============================] - 1s 29ms/step - loss: 0.6488 - acc: 0.7847
Epoch 14/100
25/25 [==============================] - 1s 30ms/step - loss: 0.6547 - acc: 0.8022
Epoch 15/100
25/25 [==============================] - 1s 29ms/step - loss: 0.5985 - acc: 0.8182
Epoch 16/100
25/25 [==============================] - 1s 29ms/step - loss: 0.5406 - acc: 0.8483
Epoch 17/100
25/25 [==============================] - 1s 29ms/step - loss: 0.5002 - acc: 0.8703
Epoch 18/100
25/25 [==============================] - 1s 29ms/step - loss: 0.4994 - acc: 0.8855
Epoch 19/100
25/25 [==============================] -

25/25 [==============================] - 1s 28ms/step - loss: 0.2381 - acc: 0.9814
Epoch 95/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2375 - acc: 0.9816
Epoch 96/100
25/25 [==============================] - 1s 27ms/step - loss: 0.2385 - acc: 0.9827
Epoch 97/100
25/25 [==============================] - 1s 27ms/step - loss: 0.8236 - acc: 0.9055
Epoch 98/100
25/25 [==============================] - 1s 27ms/step - loss: 0.6356 - acc: 0.8907
Epoch 99/100
25/25 [==============================] - 1s 28ms/step - loss: 0.4899 - acc: 0.9296
Epoch 100/100
25/25 [==============================] - 1s 28ms/step - loss: 0.4625 - acc: 0.9303
Epoch 1/100
25/25 [==============================] - 3s 29ms/step - loss: 2.0044 - acc: 0.5162
Epoch 2/100
25/25 [==============================] - 1s 29ms/step - loss: 0.9490 - acc: 0.6111
Epoch 3/100
25/25 [==============================] - 1s 29ms/step - loss: 0.7872 - acc: 0.6799
Epoch 4/100
25/25 [==============================] - 1s

25/25 [==============================] - 1s 30ms/step - loss: 0.2223 - acc: 0.9806
Epoch 77/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2159 - acc: 0.9817
Epoch 78/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3192 - acc: 0.9678
Epoch 79/100
25/25 [==============================] - 1s 29ms/step - loss: 0.4601 - acc: 0.9201
Epoch 80/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3801 - acc: 0.9348
Epoch 81/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3031 - acc: 0.9492
Epoch 82/100
25/25 [==============================] - 1s 30ms/step - loss: 0.5442 - acc: 0.9112
Epoch 83/100
25/25 [==============================] - 1s 30ms/step - loss: 0.3810 - acc: 0.9419
Epoch 84/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3100 - acc: 0.9615
Epoch 85/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2757 - acc: 0.9698
Epoch 86/100
25/25 [==============================] -

25/25 [==============================] - 1s 29ms/step - loss: 0.3295 - acc: 0.9791
Epoch 59/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3146 - acc: 0.9804
Epoch 60/100
25/25 [==============================] - 1s 28ms/step - loss: 0.4027 - acc: 0.9654
Epoch 61/100
25/25 [==============================] - 1s 29ms/step - loss: 1.0153 - acc: 0.8437
Epoch 62/100
25/25 [==============================] - 1s 28ms/step - loss: 0.5124 - acc: 0.9059
Epoch 63/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3932 - acc: 0.9531
Epoch 64/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3441 - acc: 0.9710
Epoch 65/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3136 - acc: 0.9738
Epoch 66/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2949 - acc: 0.9777
Epoch 67/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2947 - acc: 0.9785
Epoch 68/100
25/25 [==============================] -

25/25 [==============================] - 1s 29ms/step - loss: 0.2893 - acc: 0.9689
Epoch 41/100
25/25 [==============================] - 1s 28ms/step - loss: 0.2645 - acc: 0.9743
Epoch 42/100
25/25 [==============================] - 1s 30ms/step - loss: 0.2534 - acc: 0.9759
Epoch 43/100
25/25 [==============================] - 1s 38ms/step - loss: 0.2470 - acc: 0.9753
Epoch 44/100
25/25 [==============================] - 1s 31ms/step - loss: 0.2269 - acc: 0.9806
Epoch 45/100
25/25 [==============================] - 1s 30ms/step - loss: 0.2188 - acc: 0.9790
Epoch 46/100
25/25 [==============================] - 1s 30ms/step - loss: 0.2399 - acc: 0.9778
Epoch 47/100
25/25 [==============================] - 1s 31ms/step - loss: 0.6426 - acc: 0.9471
Epoch 48/100
25/25 [==============================] - 1s 31ms/step - loss: 1.8875 - acc: 0.7460
Epoch 49/100
25/25 [==============================] - 1s 30ms/step - loss: 0.6140 - acc: 0.8222
Epoch 50/100
25/25 [==============================] -

25/25 [==============================] - 1s 30ms/step - loss: 0.4035 - acc: 0.9432
Epoch 23/100
25/25 [==============================] - 1s 30ms/step - loss: 0.3677 - acc: 0.9487
Epoch 24/100
25/25 [==============================] - 1s 31ms/step - loss: 0.5105 - acc: 0.9196
Epoch 25/100
25/25 [==============================] - 1s 31ms/step - loss: 0.4262 - acc: 0.9245
Epoch 26/100
25/25 [==============================] - 1s 30ms/step - loss: 0.3756 - acc: 0.9450
Epoch 27/100
25/25 [==============================] - 1s 30ms/step - loss: 0.3443 - acc: 0.9563
Epoch 28/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3518 - acc: 0.9531
Epoch 29/100
25/25 [==============================] - 1s 30ms/step - loss: 0.5816 - acc: 0.9047
Epoch 30/100
25/25 [==============================] - 1s 31ms/step - loss: 0.4068 - acc: 0.9476
Epoch 31/100
25/25 [==============================] - 1s 31ms/step - loss: 0.3618 - acc: 0.9609
Epoch 32/100
25/25 [==============================] -

25/25 [==============================] - 1s 30ms/step - loss: 0.6975 - acc: 0.7669
Epoch 5/100
25/25 [==============================] - 1s 30ms/step - loss: 0.7737 - acc: 0.6731
Epoch 6/100
25/25 [==============================] - 1s 31ms/step - loss: 0.6744 - acc: 0.7553
Epoch 7/100
25/25 [==============================] - 1s 31ms/step - loss: 0.6076 - acc: 0.8093
Epoch 8/100
25/25 [==============================] - 1s 30ms/step - loss: 0.5632 - acc: 0.8232
Epoch 9/100
25/25 [==============================] - 1s 30ms/step - loss: 0.5344 - acc: 0.8384
Epoch 10/100
25/25 [==============================] - 1s 30ms/step - loss: 0.5350 - acc: 0.8483
Epoch 11/100
25/25 [==============================] - 1s 30ms/step - loss: 0.4759 - acc: 0.8699
Epoch 12/100
25/25 [==============================] - 1s 31ms/step - loss: 0.4356 - acc: 0.8949
Epoch 13/100
25/25 [==============================] - 1s 31ms/step - loss: 0.4350 - acc: 0.9084
Epoch 14/100
25/25 [==============================] - 1s 3

25/25 [==============================] - 1s 31ms/step - loss: 0.2914 - acc: 0.9764
Epoch 90/100
25/25 [==============================] - 1s 30ms/step - loss: 0.2907 - acc: 0.9777
Epoch 91/100
25/25 [==============================] - 1s 31ms/step - loss: 0.2817 - acc: 0.9804
Epoch 92/100
25/25 [==============================] - 1s 30ms/step - loss: 0.3081 - acc: 0.9699
Epoch 93/100
25/25 [==============================] - 1s 30ms/step - loss: 0.7173 - acc: 0.8860
Epoch 94/100
25/25 [==============================] - 1s 31ms/step - loss: 0.5235 - acc: 0.9044
Epoch 95/100
25/25 [==============================] - 1s 31ms/step - loss: 0.5974 - acc: 0.8525
Epoch 96/100
25/25 [==============================] - 1s 29ms/step - loss: 0.5029 - acc: 0.9029
Epoch 97/100
25/25 [==============================] - 1s 30ms/step - loss: 0.4275 - acc: 0.9448
Epoch 98/100
25/25 [==============================] - 1s 29ms/step - loss: 0.4967 - acc: 0.9393
Epoch 99/100
25/25 [==============================] -

25/25 [==============================] - 1s 31ms/step - loss: 0.4546 - acc: 0.9351
Epoch 72/100
25/25 [==============================] - 1s 30ms/step - loss: 0.3908 - acc: 0.9576
Epoch 73/100
25/25 [==============================] - 1s 30ms/step - loss: 0.3488 - acc: 0.9654
Epoch 74/100
25/25 [==============================] - 1s 31ms/step - loss: 0.4472 - acc: 0.9366
Epoch 75/100
25/25 [==============================] - 1s 32ms/step - loss: 0.4665 - acc: 0.9408
Epoch 76/100
25/25 [==============================] - 1s 31ms/step - loss: 0.3982 - acc: 0.9528
Epoch 77/100
25/25 [==============================] - 1s 31ms/step - loss: 0.3587 - acc: 0.9662
Epoch 78/100
25/25 [==============================] - 1s 30ms/step - loss: 0.3329 - acc: 0.9733
Epoch 79/100
25/25 [==============================] - 1s 31ms/step - loss: 0.3158 - acc: 0.9775
Epoch 80/100
25/25 [==============================] - 1s 30ms/step - loss: 0.2994 - acc: 0.9798
Epoch 81/100
25/25 [==============================] -

25/25 [==============================] - 1s 28ms/step - loss: 0.2520 - acc: 0.9793
Epoch 54/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2424 - acc: 0.9804
Epoch 55/100
25/25 [==============================] - 1s 30ms/step - loss: 0.2377 - acc: 0.9811
Epoch 56/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2464 - acc: 0.9786
Epoch 57/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2516 - acc: 0.9782
Epoch 58/100
25/25 [==============================] - 1s 30ms/step - loss: 0.2395 - acc: 0.9795
Epoch 59/100
25/25 [==============================] - 1s 39ms/step - loss: 0.8714 - acc: 0.8460
Epoch 60/100
25/25 [==============================] - 1s 31ms/step - loss: 0.6243 - acc: 0.8253
Epoch 61/100
25/25 [==============================] - 1s 30ms/step - loss: 0.4449 - acc: 0.9154
Epoch 62/100
25/25 [==============================] - 1s 30ms/step - loss: 0.3629 - acc: 0.9414
Epoch 63/100
25/25 [==============================] -

25/25 [==============================] - 1s 31ms/step - loss: 0.3641 - acc: 0.9573
Epoch 36/100
25/25 [==============================] - 1s 31ms/step - loss: 0.3817 - acc: 0.9497
Epoch 37/100
25/25 [==============================] - 1s 31ms/step - loss: 0.4269 - acc: 0.9215
Epoch 38/100
25/25 [==============================] - 1s 30ms/step - loss: 1.0916 - acc: 0.7776
Epoch 39/100
25/25 [==============================] - 1s 31ms/step - loss: 0.6047 - acc: 0.8554
Epoch 40/100
25/25 [==============================] - 1s 31ms/step - loss: 0.4934 - acc: 0.9169
Epoch 41/100
25/25 [==============================] - 1s 31ms/step - loss: 0.4591 - acc: 0.9193
Epoch 42/100
25/25 [==============================] - 1s 31ms/step - loss: 0.5740 - acc: 0.8698
Epoch 43/100
25/25 [==============================] - 1s 31ms/step - loss: 0.5527 - acc: 0.8970
Epoch 44/100
25/25 [==============================] - 1s 31ms/step - loss: 0.4665 - acc: 0.9240
Epoch 45/100
25/25 [==============================] -

25/25 [==============================] - 1s 30ms/step - loss: 0.6587 - acc: 0.7913
Epoch 18/100
25/25 [==============================] - 1s 32ms/step - loss: 0.6289 - acc: 0.8128
Epoch 19/100
25/25 [==============================] - 1s 30ms/step - loss: 0.5614 - acc: 0.8619
Epoch 20/100
25/25 [==============================] - 1s 30ms/step - loss: 0.5090 - acc: 0.8716
Epoch 21/100
25/25 [==============================] - 1s 30ms/step - loss: 0.4745 - acc: 0.8908
Epoch 22/100
25/25 [==============================] - 1s 30ms/step - loss: 0.4600 - acc: 0.9038
Epoch 23/100
25/25 [==============================] - 1s 31ms/step - loss: 0.4612 - acc: 0.9023
Epoch 24/100
25/25 [==============================] - 1s 31ms/step - loss: 0.4773 - acc: 0.9041
Epoch 25/100
25/25 [==============================] - 1s 31ms/step - loss: 0.4045 - acc: 0.9246
Epoch 26/100
25/25 [==============================] - 1s 31ms/step - loss: 0.3786 - acc: 0.9301
Epoch 27/100
25/25 [==============================] -

Epoch 1/100
25/25 [==============================] - 3s 29ms/step - loss: 1.9422 - acc: 0.5191
Epoch 2/100
25/25 [==============================] - 1s 29ms/step - loss: 1.0613 - acc: 0.5953
Epoch 3/100
25/25 [==============================] - 1s 29ms/step - loss: 0.8924 - acc: 0.6574
Epoch 4/100
25/25 [==============================] - 1s 28ms/step - loss: 0.7817 - acc: 0.7543
Epoch 5/100
25/25 [==============================] - 1s 30ms/step - loss: 0.7666 - acc: 0.7417
Epoch 6/100
25/25 [==============================] - 1s 29ms/step - loss: 0.7556 - acc: 0.7795
Epoch 7/100
25/25 [==============================] - 1s 28ms/step - loss: 0.6869 - acc: 0.7889
Epoch 8/100
25/25 [==============================] - 1s 30ms/step - loss: 0.6528 - acc: 0.8124
Epoch 9/100
25/25 [==============================] - 1s 29ms/step - loss: 0.6222 - acc: 0.8323
Epoch 10/100
25/25 [==============================] - 1s 29ms/step - loss: 0.6112 - acc: 0.8340
Epoch 11/100
25/25 [=============================

25/25 [==============================] - 1s 29ms/step - loss: 0.2938 - acc: 0.9775
Epoch 85/100
25/25 [==============================] - 1s 28ms/step - loss: 0.3772 - acc: 0.9589
Epoch 86/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3221 - acc: 0.9735
Epoch 87/100
25/25 [==============================] - 1s 30ms/step - loss: 0.6939 - acc: 0.9403
Epoch 88/100
25/25 [==============================] - 1s 29ms/step - loss: 0.8600 - acc: 0.8488
Epoch 89/100
25/25 [==============================] - 1s 30ms/step - loss: 0.3909 - acc: 0.9283
Epoch 90/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3249 - acc: 0.9602
Epoch 91/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2894 - acc: 0.9723
Epoch 92/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2654 - acc: 0.9783
Epoch 93/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3211 - acc: 0.9615
Epoch 94/100
25/25 [==============================] -

25/25 [==============================] - 1s 30ms/step - loss: 0.3404 - acc: 0.9709
Epoch 66/100
25/25 [==============================] - 1s 28ms/step - loss: 0.2992 - acc: 0.9799
Epoch 67/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2854 - acc: 0.9814
Epoch 68/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2802 - acc: 0.9809
Epoch 69/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2688 - acc: 0.9827
Epoch 70/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2767 - acc: 0.9827
Epoch 71/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2618 - acc: 0.9845
Epoch 72/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2580 - acc: 0.9850
Epoch 73/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2517 - acc: 0.9864
Epoch 74/100
25/25 [==============================] - 1s 30ms/step - loss: 0.2483 - acc: 0.9872
Epoch 75/100
25/25 [==============================] -

25/25 [==============================] - 1s 29ms/step - loss: 0.2770 - acc: 0.9764
Epoch 48/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2598 - acc: 0.9806
Epoch 49/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2520 - acc: 0.9832
Epoch 50/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2493 - acc: 0.9812
Epoch 51/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2378 - acc: 0.9829
Epoch 52/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2307 - acc: 0.9856
Epoch 53/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2630 - acc: 0.9811
Epoch 54/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3402 - acc: 0.9657
Epoch 55/100
25/25 [==============================] - 1s 29ms/step - loss: 0.9418 - acc: 0.8248
Epoch 56/100
25/25 [==============================] - 1s 29ms/step - loss: 0.6772 - acc: 0.8138
Epoch 57/100
25/25 [==============================] -

25/25 [==============================] - 1s 28ms/step - loss: 0.4084 - acc: 0.9293
Epoch 30/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3729 - acc: 0.9442
Epoch 31/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3495 - acc: 0.9518
Epoch 32/100
25/25 [==============================] - 1s 29ms/step - loss: 0.7270 - acc: 0.8293
Epoch 33/100
25/25 [==============================] - 1s 29ms/step - loss: 0.5814 - acc: 0.8424
Epoch 34/100
25/25 [==============================] - 1s 29ms/step - loss: 0.4768 - acc: 0.9173
Epoch 35/100
25/25 [==============================] - 1s 29ms/step - loss: 0.4158 - acc: 0.9426
Epoch 36/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3978 - acc: 0.9437
Epoch 37/100
25/25 [==============================] - 1s 28ms/step - loss: 0.3749 - acc: 0.9481
Epoch 38/100
25/25 [==============================] - 1s 29ms/step - loss: 0.4629 - acc: 0.9055
Epoch 39/100
25/25 [==============================] -

25/25 [==============================] - 1s 28ms/step - loss: 0.5909 - acc: 0.8248
Epoch 12/100
25/25 [==============================] - 1s 29ms/step - loss: 0.7921 - acc: 0.6281
Epoch 13/100
25/25 [==============================] - 1s 29ms/step - loss: 0.8104 - acc: 0.6000
Epoch 14/100
25/25 [==============================] - 1s 28ms/step - loss: 0.6691 - acc: 0.7884
Epoch 15/100
25/25 [==============================] - 1s 29ms/step - loss: 0.6209 - acc: 0.8162
Epoch 16/100
25/25 [==============================] - 1s 29ms/step - loss: 0.6174 - acc: 0.8200
Epoch 17/100
25/25 [==============================] - 1s 29ms/step - loss: 0.5677 - acc: 0.8442
Epoch 18/100
25/25 [==============================] - 1s 29ms/step - loss: 0.5252 - acc: 0.8610
Epoch 19/100
25/25 [==============================] - 1s 28ms/step - loss: 0.4998 - acc: 0.8712
Epoch 20/100
25/25 [==============================] - 1s 28ms/step - loss: 0.4916 - acc: 0.8858
Epoch 21/100
25/25 [==============================] -

25/25 [==============================] - 1s 27ms/step - loss: 0.3780 - acc: 0.9560
Epoch 97/100
25/25 [==============================] - 1s 28ms/step - loss: 0.3817 - acc: 0.9605
Epoch 98/100
25/25 [==============================] - 1s 28ms/step - loss: 0.4391 - acc: 0.9588
Epoch 99/100
25/25 [==============================] - 1s 27ms/step - loss: 0.3971 - acc: 0.9615
Epoch 100/100
25/25 [==============================] - 1s 27ms/step - loss: 0.3815 - acc: 0.9657
Epoch 1/100
25/25 [==============================] - 3s 29ms/step - loss: 1.7691 - acc: 0.4942
Epoch 2/100
25/25 [==============================] - 1s 29ms/step - loss: 1.0079 - acc: 0.5984
Epoch 3/100
25/25 [==============================] - 1s 29ms/step - loss: 0.8253 - acc: 0.6797
Epoch 4/100
25/25 [==============================] - 1s 29ms/step - loss: 0.7327 - acc: 0.7488
Epoch 5/100
25/25 [==============================] - 1s 29ms/step - loss: 0.6780 - acc: 0.7897
Epoch 6/100
25/25 [==============================] - 1s 3

25/25 [==============================] - 1s 29ms/step - loss: 0.2066 - acc: 0.9893
Epoch 79/100
25/25 [==============================] - 1s 30ms/step - loss: 0.2335 - acc: 0.9867
Epoch 80/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2442 - acc: 0.9840
Epoch 81/100
25/25 [==============================] - 1s 30ms/step - loss: 0.3248 - acc: 0.9717
Epoch 82/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3365 - acc: 0.9719
Epoch 83/100
25/25 [==============================] - 1s 30ms/step - loss: 0.3212 - acc: 0.9767
Epoch 84/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3434 - acc: 0.9660
Epoch 85/100
25/25 [==============================] - 1s 29ms/step - loss: 0.4199 - acc: 0.9516
Epoch 86/100
25/25 [==============================] - 1s 30ms/step - loss: 0.3614 - acc: 0.9621
Epoch 87/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2864 - acc: 0.9795
Epoch 88/100
25/25 [==============================] -

25/25 [==============================] - 1s 29ms/step - loss: 0.6766 - acc: 0.8188
Epoch 61/100
25/25 [==============================] - 1s 29ms/step - loss: 0.6644 - acc: 0.8465
Epoch 62/100
25/25 [==============================] - 1s 29ms/step - loss: 0.6068 - acc: 0.8352
Epoch 63/100
25/25 [==============================] - 1s 29ms/step - loss: 0.5581 - acc: 0.8646
Epoch 64/100
25/25 [==============================] - 1s 30ms/step - loss: 0.5106 - acc: 0.8992
Epoch 65/100
25/25 [==============================] - 1s 29ms/step - loss: 0.5000 - acc: 0.9136
Epoch 66/100
25/25 [==============================] - 1s 29ms/step - loss: 0.4714 - acc: 0.9177
Epoch 67/100
25/25 [==============================] - 1s 30ms/step - loss: 0.4306 - acc: 0.9264
Epoch 68/100
25/25 [==============================] - 1s 29ms/step - loss: 0.4300 - acc: 0.9346
Epoch 69/100
25/25 [==============================] - 1s 29ms/step - loss: 0.4266 - acc: 0.9338
Epoch 70/100
25/25 [==============================] -

25/25 [==============================] - 1s 29ms/step - loss: 0.5345 - acc: 0.8894
Epoch 43/100
25/25 [==============================] - 1s 29ms/step - loss: 0.4013 - acc: 0.9340
Epoch 44/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3406 - acc: 0.9578
Epoch 45/100
25/25 [==============================] - 1s 30ms/step - loss: 0.2900 - acc: 0.9685
Epoch 46/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2739 - acc: 0.9733
Epoch 47/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2714 - acc: 0.9698
Epoch 48/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2645 - acc: 0.9733
Epoch 49/100
25/25 [==============================] - 1s 28ms/step - loss: 0.2556 - acc: 0.9782
Epoch 50/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2394 - acc: 0.9788
Epoch 51/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2277 - acc: 0.9816
Epoch 52/100
25/25 [==============================] -

25/25 [==============================] - 1s 28ms/step - loss: 0.4183 - acc: 0.9225
Epoch 25/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3722 - acc: 0.9350
Epoch 26/100
25/25 [==============================] - 1s 28ms/step - loss: 0.3616 - acc: 0.9389
Epoch 27/100
25/25 [==============================] - 1s 28ms/step - loss: 0.3607 - acc: 0.9350
Epoch 28/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3088 - acc: 0.9526
Epoch 29/100
25/25 [==============================] - 1s 28ms/step - loss: 0.3280 - acc: 0.9463
Epoch 30/100
25/25 [==============================] - 1s 28ms/step - loss: 0.3130 - acc: 0.9529
Epoch 31/100
25/25 [==============================] - 1s 28ms/step - loss: 0.2828 - acc: 0.9615
Epoch 32/100
25/25 [==============================] - 1s 28ms/step - loss: 0.2614 - acc: 0.9665
Epoch 33/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2480 - acc: 0.9699
Epoch 34/100
25/25 [==============================] -

25/25 [==============================] - 1s 29ms/step - loss: 0.7522 - acc: 0.7312
Epoch 7/100
25/25 [==============================] - 1s 29ms/step - loss: 0.7150 - acc: 0.7637
Epoch 8/100
25/25 [==============================] - 1s 29ms/step - loss: 0.6832 - acc: 0.7905
Epoch 9/100
25/25 [==============================] - 1s 29ms/step - loss: 0.6435 - acc: 0.8128
Epoch 10/100
25/25 [==============================] - 1s 30ms/step - loss: 0.6051 - acc: 0.8342
Epoch 11/100
25/25 [==============================] - 1s 29ms/step - loss: 0.5792 - acc: 0.8445
Epoch 12/100
25/25 [==============================] - 1s 29ms/step - loss: 0.5755 - acc: 0.8467
Epoch 13/100
25/25 [==============================] - 1s 29ms/step - loss: 0.6087 - acc: 0.8149
Epoch 14/100
25/25 [==============================] - 1s 28ms/step - loss: 0.5611 - acc: 0.8662
Epoch 15/100
25/25 [==============================] - 1s 29ms/step - loss: 0.5068 - acc: 0.8903
Epoch 16/100
25/25 [==============================] - 1s

25/25 [==============================] - 1s 29ms/step - loss: 0.3373 - acc: 0.9712
Epoch 92/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3818 - acc: 0.9599
Epoch 93/100
25/25 [==============================] - 1s 28ms/step - loss: 0.5666 - acc: 0.9241
Epoch 94/100
25/25 [==============================] - 1s 30ms/step - loss: 0.3859 - acc: 0.9497
Epoch 95/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3599 - acc: 0.9630
Epoch 96/100
25/25 [==============================] - 1s 27ms/step - loss: 0.3269 - acc: 0.9719
Epoch 97/100
25/25 [==============================] - 1s 27ms/step - loss: 0.3172 - acc: 0.9736
Epoch 98/100
25/25 [==============================] - 1s 27ms/step - loss: 0.3395 - acc: 0.9743
Epoch 99/100
25/25 [==============================] - 1s 28ms/step - loss: 0.4057 - acc: 0.9639
Epoch 100/100
25/25 [==============================] - 1s 27ms/step - loss: 0.4113 - acc: 0.9618
Epoch 1/100
25/25 [==============================] -

25/25 [==============================] - 1s 29ms/step - loss: 0.3737 - acc: 0.9558
Epoch 74/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3528 - acc: 0.9626
Epoch 75/100
25/25 [==============================] - 1s 28ms/step - loss: 0.3365 - acc: 0.9699
Epoch 76/100
25/25 [==============================] - 1s 28ms/step - loss: 0.3301 - acc: 0.9753
Epoch 77/100
25/25 [==============================] - 1s 28ms/step - loss: 0.3073 - acc: 0.9786
Epoch 78/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2904 - acc: 0.9806
Epoch 79/100
25/25 [==============================] - 1s 29ms/step - loss: 0.9466 - acc: 0.8835
Epoch 80/100
25/25 [==============================] - 1s 29ms/step - loss: 0.5302 - acc: 0.8942
Epoch 81/100
25/25 [==============================] - 1s 29ms/step - loss: 0.5025 - acc: 0.9282
Epoch 82/100
25/25 [==============================] - 1s 28ms/step - loss: 0.4936 - acc: 0.8850
Epoch 83/100
25/25 [==============================] -

25/25 [==============================] - 1s 28ms/step - loss: 0.4802 - acc: 0.8987
Epoch 56/100
25/25 [==============================] - 1s 28ms/step - loss: 0.4993 - acc: 0.9201
Epoch 57/100
25/25 [==============================] - 1s 28ms/step - loss: 0.4879 - acc: 0.9152
Epoch 58/100
25/25 [==============================] - 1s 29ms/step - loss: 0.4561 - acc: 0.9245
Epoch 59/100
25/25 [==============================] - 1s 29ms/step - loss: 0.4301 - acc: 0.9363
Epoch 60/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3914 - acc: 0.9471
Epoch 61/100
25/25 [==============================] - 1s 28ms/step - loss: 0.4026 - acc: 0.9466
Epoch 62/100
25/25 [==============================] - 1s 29ms/step - loss: 0.6157 - acc: 0.8824
Epoch 63/100
25/25 [==============================] - 1s 29ms/step - loss: 0.5081 - acc: 0.8955
Epoch 64/100
25/25 [==============================] - 1s 30ms/step - loss: 0.4170 - acc: 0.9313
Epoch 65/100
25/25 [==============================] -

25/25 [==============================] - 1s 28ms/step - loss: 0.2955 - acc: 0.9621
Epoch 38/100
25/25 [==============================] - 1s 27ms/step - loss: 0.2791 - acc: 0.9657
Epoch 39/100
25/25 [==============================] - 1s 28ms/step - loss: 0.2651 - acc: 0.9683
Epoch 40/100
25/25 [==============================] - 1s 28ms/step - loss: 0.2602 - acc: 0.9688
Epoch 41/100
25/25 [==============================] - 1s 28ms/step - loss: 0.3330 - acc: 0.9638
Epoch 42/100
25/25 [==============================] - 1s 28ms/step - loss: 1.2916 - acc: 0.8852
Epoch 43/100
25/25 [==============================] - 1s 28ms/step - loss: 2.4318 - acc: 0.7138
Epoch 44/100
25/25 [==============================] - 1s 28ms/step - loss: 0.8632 - acc: 0.7760
Epoch 45/100
25/25 [==============================] - 1s 28ms/step - loss: 0.6976 - acc: 0.8397
Epoch 46/100
25/25 [==============================] - 1s 29ms/step - loss: 0.5436 - acc: 0.8916
Epoch 47/100
25/25 [==============================] -

25/25 [==============================] - 1s 29ms/step - loss: 0.5314 - acc: 0.8976
Epoch 20/100
25/25 [==============================] - 1s 28ms/step - loss: 0.4686 - acc: 0.9081
Epoch 21/100
25/25 [==============================] - 1s 30ms/step - loss: 0.4246 - acc: 0.9256
Epoch 22/100
25/25 [==============================] - 1s 29ms/step - loss: 0.9520 - acc: 0.8763
Epoch 23/100
25/25 [==============================] - 1s 29ms/step - loss: 0.7267 - acc: 0.8780
Epoch 24/100
25/25 [==============================] - 1s 29ms/step - loss: 0.5222 - acc: 0.9059
Epoch 25/100
25/25 [==============================] - 1s 29ms/step - loss: 0.5070 - acc: 0.9065
Epoch 26/100
25/25 [==============================] - 1s 29ms/step - loss: 0.4348 - acc: 0.9295
Epoch 27/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3935 - acc: 0.9384
Epoch 28/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3746 - acc: 0.9431
Epoch 29/100
25/25 [==============================] -

25/25 [==============================] - 2s 29ms/step - loss: 1.9854 - acc: 0.5157
Epoch 2/100
25/25 [==============================] - 1s 28ms/step - loss: 0.9209 - acc: 0.6313
Epoch 3/100
25/25 [==============================] - 1s 28ms/step - loss: 0.7698 - acc: 0.7062
Epoch 4/100
25/25 [==============================] - 1s 28ms/step - loss: 0.6830 - acc: 0.7609
Epoch 5/100
25/25 [==============================] - 1s 29ms/step - loss: 0.6330 - acc: 0.8056
Epoch 6/100
25/25 [==============================] - 1s 29ms/step - loss: 0.6003 - acc: 0.8259
Epoch 7/100
25/25 [==============================] - 1s 28ms/step - loss: 0.5754 - acc: 0.8481
Epoch 8/100
25/25 [==============================] - 1s 28ms/step - loss: 0.5722 - acc: 0.8580
Epoch 9/100
25/25 [==============================] - 1s 29ms/step - loss: 0.5872 - acc: 0.8649
Epoch 10/100
25/25 [==============================] - 1s 28ms/step - loss: 0.5211 - acc: 0.8767
Epoch 11/100
25/25 [==============================] - 1s 28ms

25/25 [==============================] - 1s 30ms/step - loss: 0.2622 - acc: 0.9858
Epoch 87/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3531 - acc: 0.9675
Epoch 88/100
25/25 [==============================] - 1s 29ms/step - loss: 0.5592 - acc: 0.9379
Epoch 89/100
25/25 [==============================] - 1s 29ms/step - loss: 0.4122 - acc: 0.9476
Epoch 90/100
25/25 [==============================] - 1s 29ms/step - loss: 0.4753 - acc: 0.9508
Epoch 91/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3768 - acc: 0.9581
Epoch 92/100
25/25 [==============================] - 1s 30ms/step - loss: 0.3215 - acc: 0.9762
Epoch 93/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2831 - acc: 0.9835
Epoch 94/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2645 - acc: 0.9858
Epoch 95/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2531 - acc: 0.9879
Epoch 96/100
25/25 [==============================] -

25/25 [==============================] - 1s 30ms/step - loss: 0.4898 - acc: 0.8989
Epoch 69/100
25/25 [==============================] - 1s 28ms/step - loss: 0.4434 - acc: 0.9272
Epoch 70/100
25/25 [==============================] - 1s 28ms/step - loss: 0.4114 - acc: 0.9406
Epoch 71/100
25/25 [==============================] - 1s 28ms/step - loss: 0.5365 - acc: 0.8848
Epoch 72/100
25/25 [==============================] - 1s 28ms/step - loss: 0.4160 - acc: 0.9306
Epoch 73/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3779 - acc: 0.9489
Epoch 74/100
25/25 [==============================] - 1s 29ms/step - loss: 0.5899 - acc: 0.8619
Epoch 75/100
25/25 [==============================] - 1s 28ms/step - loss: 0.4954 - acc: 0.8577
Epoch 76/100
25/25 [==============================] - 1s 29ms/step - loss: 0.4330 - acc: 0.9029
Epoch 77/100
25/25 [==============================] - 1s 29ms/step - loss: 0.4337 - acc: 0.9138
Epoch 78/100
25/25 [==============================] -

25/25 [==============================] - 1s 29ms/step - loss: 0.2311 - acc: 0.9850
Epoch 51/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2227 - acc: 0.9861
Epoch 52/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3866 - acc: 0.9667
Epoch 53/100
25/25 [==============================] - 1s 29ms/step - loss: 0.4722 - acc: 0.9397
Epoch 54/100
25/25 [==============================] - 1s 30ms/step - loss: 1.6384 - acc: 0.7884
Epoch 55/100
25/25 [==============================] - 1s 29ms/step - loss: 0.5803 - acc: 0.8698
Epoch 56/100
25/25 [==============================] - 1s 30ms/step - loss: 0.4464 - acc: 0.9361
Epoch 57/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3715 - acc: 0.9578
Epoch 58/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3308 - acc: 0.9670
Epoch 59/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3208 - acc: 0.9696
Epoch 60/100
25/25 [==============================] -

25/25 [==============================] - 1s 29ms/step - loss: 0.3451 - acc: 0.9581
Epoch 33/100
25/25 [==============================] - 1s 28ms/step - loss: 0.3429 - acc: 0.9544
Epoch 34/100
25/25 [==============================] - 1s 28ms/step - loss: 0.3385 - acc: 0.9563
Epoch 35/100
25/25 [==============================] - 1s 29ms/step - loss: 0.4258 - acc: 0.9389
Epoch 36/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3964 - acc: 0.9405
Epoch 37/100
25/25 [==============================] - 1s 30ms/step - loss: 0.4175 - acc: 0.9450
Epoch 38/100
25/25 [==============================] - 1s 29ms/step - loss: 0.7388 - acc: 0.8845
Epoch 39/100
25/25 [==============================] - 1s 29ms/step - loss: 0.4397 - acc: 0.9309
Epoch 40/100
25/25 [==============================] - 1s 29ms/step - loss: 0.4053 - acc: 0.9490
Epoch 41/100
25/25 [==============================] - 1s 29ms/step - loss: 0.3388 - acc: 0.9626
Epoch 42/100
25/25 [==============================] -

25/25 [==============================] - 1s 28ms/step - loss: 0.5354 - acc: 0.8471
Epoch 15/100
25/25 [==============================] - 1s 29ms/step - loss: 0.4885 - acc: 0.8742
Epoch 16/100
25/25 [==============================] - 1s 29ms/step - loss: 0.4679 - acc: 0.8895
Epoch 17/100
25/25 [==============================] - 1s 29ms/step - loss: 0.4546 - acc: 0.8963
Epoch 18/100
25/25 [==============================] - 1s 29ms/step - loss: 0.4741 - acc: 0.8965
Epoch 19/100
25/25 [==============================] - 1s 29ms/step - loss: 0.4520 - acc: 0.9088
Epoch 20/100
25/25 [==============================] - 1s 28ms/step - loss: 0.4145 - acc: 0.9185
Epoch 21/100
25/25 [==============================] - 1s 28ms/step - loss: 0.4319 - acc: 0.9240
Epoch 22/100
25/25 [==============================] - 1s 29ms/step - loss: 0.7275 - acc: 0.8225
Epoch 23/100
25/25 [==============================] - 1s 29ms/step - loss: 0.5804 - acc: 0.8153
Epoch 24/100
25/25 [==============================] -

25/25 [==============================] - 1s 27ms/step - loss: 0.2219 - acc: 0.9861
Epoch 100/100
25/25 [==============================] - 1s 26ms/step - loss: 0.2623 - acc: 0.9769
Epoch 1/100
25/25 [==============================] - 2s 28ms/step - loss: 1.9337 - acc: 0.5089
Epoch 2/100
25/25 [==============================] - 1s 28ms/step - loss: 1.0133 - acc: 0.5987
Epoch 3/100
25/25 [==============================] - 1s 28ms/step - loss: 0.8859 - acc: 0.6457
Epoch 4/100
25/25 [==============================] - 1s 28ms/step - loss: 0.8192 - acc: 0.6957
Epoch 5/100
25/25 [==============================] - 1s 28ms/step - loss: 0.7232 - acc: 0.7688
Epoch 6/100
25/25 [==============================] - 1s 29ms/step - loss: 0.6847 - acc: 0.7871
Epoch 7/100
25/25 [==============================] - 1s 29ms/step - loss: 0.7018 - acc: 0.7900
Epoch 8/100
25/25 [==============================] - 1s 28ms/step - loss: 0.7894 - acc: 0.6876
Epoch 9/100
25/25 [==============================] - 1s 28ms

25/25 [==============================] - 1s 28ms/step - loss: 0.2210 - acc: 0.9864
Epoch 82/100
25/25 [==============================] - 1s 29ms/step - loss: 0.2177 - acc: 0.9884
Epoch 83/100
25/25 [==============================] - 1s 28ms/step - loss: 0.2168 - acc: 0.9872
Epoch 84/100
25/25 [==============================] - 1s 28ms/step - loss: 0.2306 - acc: 0.9840
Epoch 85/100
25/25 [==============================] - 1s 28ms/step - loss: 0.2527 - acc: 0.9783
Epoch 86/100
25/25 [==============================] - 1s 28ms/step - loss: 0.2763 - acc: 0.9749
Epoch 87/100
25/25 [==============================] - 1s 28ms/step - loss: 1.5752 - acc: 0.8462
Epoch 88/100
25/25 [==============================] - 1s 28ms/step - loss: 0.8542 - acc: 0.8373
Epoch 89/100
25/25 [==============================] - 1s 28ms/step - loss: 0.6446 - acc: 0.8596
Epoch 90/100
25/25 [==============================] - 1s 28ms/step - loss: 0.5209 - acc: 0.9012
Epoch 91/100
25/25 [==============================] -

In [16]:
from sklearn.metrics import accuracy_score
from sklearn.metrics import precision_score
from sklearn.metrics import recall_score
from sklearn.metrics import f1_score
from sklearn.metrics import roc_auc_score
preds = final_model.predict(X_test)

precision = precision_score(y_test, preds)
recall = recall_score(y_test, preds)
f1 = f1_score(y_test, preds)
roc_auc = roc_auc_score(y_test, preds)
acc = accuracy_score(y_test, preds)

print(f' accuracy : {acc}, precision : {precision}, recall : {recall}, f1 : {f1}, roc_auc : {roc_auc}')

 accuracy : 0.7488673139158576, precision : 0.7477064220183486, recall : 0.8782327586206896, f1 : 0.8077304261645193, roc_auc : 0.7162638671547533


**예전 LSTM 구조 acc**
- `KerasClassifier(build_fn=lambda:get_model(), epochs=20, batch_size=256)`
  - 1개 : 0.7223300970873786
  - 5개 : 0.7436893203883496
  - 15개 : 0.7372168284789644
  - 25개 : 0.7404530744336569
  - 50개 : 0.7411003236245954
- `KerasClassifier(build_fn=lambda:get_model(), epochs=100, batch_size=256)`
  - 1개 : 0.7313915857605178
  - 5개 : 0.7449838187702266
  - 15개 : { accuracy : 0.7585760517799353, precision : 0.7707317073170732, recall : 0.8512931034482759, f1 : 0.8090117767537122, roc_auc : 0.7352089504275415 }
  - 25개 : { accuracy : 0.7592233009708738, precision : 0.7796780684104627, recall : 0.8351293103448276, f1 : 0.8064516129032258, roc_auc : 0.7400930182194154 }
  - 50개 : accuracy : 0.7488673139158576, precision : 0.7477064220183486, recall : 0.8782327586206896, f1 : 0.8077304261645193, roc_auc : 0.7162638671547533